### PR TITLE
Add oneOf metadata 

### PR DESCRIPTION
**Pull request by IngaFeick**
Introducing a new config option `pb3_set_oneof_metainfo` which adds meta information to `[@metadata][pb_oneof]` about which classes were chosen for [oneof](https://developers.google.com/protocol-buffers/docs/proto3#oneof) fields.
Example values: for the protobuf definition
```
    oneof :horse_type do
      optional :unicorn, :message, 2, "UnicornType"
      optional :pegasus, :message, 3, "PegasusType"
    end
```
the field `[@metadata][pb_oneof][horse_type]` will be set to either `pegasus` or `unicorn`.
Available only for protobuf version 3. The option is deactivated by default. 
Version bump to 1.2.3.

### CODE REVIEW

File: lib/logstash/codecs/protobuf.rb [hunk 1]
```diff
@@ -297,8 +313,8 @@ def pb3_handle_type_errors(event, e, is_recursive_call, datahash)
           @logger.warn(msg)
           mismatches = pb3_get_type_mismatches(datahash, "", @class_name)
 
-          msg = "Protobuf encoding info 2.2: Type mismatches found: #{mismatches}." # TODO remove
-          @logger.warn(msg)
+          #msg = "Protobuf encoding info 2.2: Type mismatches found: #{mismatches}."
```
Comment (jorgelbg) [inline lib/logstash/codecs/protobuf.rb]: Should we remove it entirely?

File: docs/index.asciidoc [hunk 2]
```diff
@@ -207,8 +207,31 @@ Stop entire pipeline when encountering a non decodable message.
...
+
+Available only for protobuf version 3.
+
+[id="plugins-{type}s-{plugin}-pb3_set_oneof_metainfo"]
+===== `pb3_set_oneof_metainfo`
+
+  * Value type is <<boolean,boolean>>
+  * Default value is false
+
+Add meta information to `[@metadata][pb_oneof]` about which classes were chosen for [oneof](https://developers.google.com/protocol-buffers/docs/proto3#oneof) fields.
```
Comment (jorgelbg) [inline docs/index.asciidoc]: nit: On the documentation perhaps we should make clear that the end value of the `[@metadata][pb_oneof]` field would look like:

```ruby
[@metadata][pb_oneof][field_name]
```

nit: Also perhaps that the field would be populated recursively for the nested fields as shown in the `pb3decoder_test8c` test.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,6 @@
+## 1.2.3
+  - Add oneof information to @metadata.
```
Comment (jorgelbg) [inline CHANGELOG.md]: Perhaps we should mention here that only it is supported for protobuf 3?

**Comment by IngaFeick** [comment]
Thanks @jorgelbg for the suggestions. PR has been updated.

**Pull request merged**